main.py

```diff
@@ -20,235 +20,278 @@
-                    if(self.right_angle > -self.anglelimit):
-                        self.right_angle -= self.anglespeed
-                    if(self.rightcar_x >= 760):
-                        self.rightmove = 0
-                        self.right = not self.right
-            # Keyboard Input
-            if event.type == pygame.KEYDOWN and event.key == 276 and self.leftclick == 0 and self.move == 1:
-                self.leftmove = 1
-                self.leftclick = 1
-
-            # left starts moving
-            if event.type == pygame.KEYUP and event.key == 276:
-                self.leftclick = 0
-            if event.type == pygame.KEYDOWN and event.key == 275 and self.rightclick == 0 and self.move == 1:
-                # jump.play(0)
-                self.rightmove = 1
-                self.rightclick = 1
-            # right start moving
-            if event.type == pygame.KEYUP and event.key == 275:
-                self.rightclick = 0
-
-            # BLACK RECTANGLES DISPLAY
-            pygame.draw.line(self.gameDisplay, black, (350, 0), (350, 768), 1)
-            pygame.draw.line(self.gameDisplay, black, (840, 0), (840, 768), 1)
-            pygame.draw.rect(self.gameDisplay, black, (0, 0, 350, 768))
-            pygame.draw.rect(self.gameDisplay, black, (840, 0, 693, 768))
-            pygame.display.update()
-            clock.tick(60)
+    def render(self):
+        self.screen.fill(BLACK)
+        self.screen.blit(self.background, (350, 0))
+        self.draw_cars()
+        self.draw_objects()
+        self.draw_score()
+
+    def draw_cars(self):
+        self.screen.blit(pygame.transform.rotate(
+            self.leftcar, self.left_angle), (self.leftcar_x, 550))
+        self.screen.blit(pygame.transform.rotate(
+            self.rightcar, self.right_angle), (self.rightcar_x, 550))
+
+    def draw_objects(self):
+        for object in self.objectlist:
+            object.display(self)
+
+    def draw_score(self):
+        score_object = self.font3.render(_(str(self.score)), 1, (WHITE))
+        self.screen.blit(score_object, (780, 20))
+
+    def handle_events(self):
+        while Gtk.events_pending():
+            Gtk.main_iteration()
+        for event in pygame.event.get():
+            self.handle_input(event)
+
+    def handle_input(self, event):
+        if event.type == pygame.QUIT:
+            self.running = False
+        if event.type == pygame.KEYDOWN:
+            if event.key == pygame.K_LEFT and not self.is_left_pressed:
+                self.left_moved = True
+                self.is_left_pressed = True
+            elif event.key == pygame.K_RIGHT and not self.is_right_pressed:
+                self.right_moved = True
+                self.is_right_pressed = True
+        elif event.type == pygame.KEYUP:
+            if event.key == pygame.K_LEFT:
+                self.is_left_pressed = False
+            elif event.key == pygame.K_RIGHT:
+                self.is_right_pressed = False
```

Comment:
This should be one function, not two separate functions.

Reply:
Agree, fixed